articles/generalized_interface_generation.md

```diff
@@ -0,0 +1,668 @@
+        },
+        "pyext_typesupport_fastrtps_c": {
+            // CPython extension to access Fast-RTPS C typesupport
+            "Type": "cpyext",
+            "Assembly": {
+                "Requires": [
+                    ":py",
+                    ":cpython",
+                    "foo_msgs_typesupport_fastrtps_c",
+                    "rosidl_runtime_c",
+                    "rosidl_typesupport_c",
+                    "rosidl_typesupport_interface",
+                    "action_msgs"
+                ],
+                "Configurations": {
+                    "GNU": {
+                        "Compiler-Flags": [ "-Wall", "-Wextra" ],
+                    }
+                    "Clang": {
+                        "Compiler-Flags": [ "-Wall", "-Wextra" ],
+                    }
+                },
+                "Sources": [ "_foo_msgs_s.ep.typesupport_fastrtps_c.c" ],
+                "Include-Requires": [ "foo_msgs_typesupport_c" ]
+            },
+            "Distribution": {
+                // Prefix should be set to the proper site-packages subdirectory
+                "Location": "@python-prefix@/foo_msgs",
+                // A proper Python extension suffix will be added for pyext components.
+                "Name": "foo_msgs_s__typesupport_fastrtps_c"
+            }
+        },
+        "pyext_typesupport_introspection_c": {
+            // CPython extension to access introspection C typesupport
+            "Type": "cpyext",
+            "Assembly": {
+                "Requires": [
+                    ":py",
+                    ":cpython",
+                    "foo_msgs_typesupport_introspection_c"
+                    "rosidl_runtime_c",
+                    "rosidl_typesupport_c",
+                    "rosidl_typesupport_interface",
+                    "action_msgs",
+                ],
+                "Configurations": {
+                    "GNU": {
+                        "Compiler-Flags": [ "-Wall", "-Wextra" ],
+                    }
+                    "Clang": {
+                        "Compiler-Flags": [ "-Wall", "-Wextra" ],
+                    }
+                },
+                "Sources": [ "_foo_msgs_s.ep.typesupport_introspection_c.c" ],
+                "Include-Requires": [ "foo_msgs_typesupport_c" ]
+            },
+            "Distribution": {
+                // Prefix should be set to the proper site-packages subdirectory
+                "Location": "@python-prefix@/foo_msgs",
+                // A proper Python extension suffix will be added for pyext components
+                "Name": "foo_msgs_s__typesupport_introspection_c"
+            }
+        }
+    }
+}
+```
+
+A few details worth noting:
+
+- The explicit dependency on other `foo_msgs_*` generated packages, which the tooling should resolve if all are to be present on the same specification.
+- The requirement on the `python-dev` package, which is a hint for the build system to build against CPython development libraries.
```

Comment:
TL; DR not a `rosdep` key, just a package this build specification depends on.

Generally speaking, a package requirement abstracts away a dependency. A package may exist for the build system, or it may not i.e. it may be replaced by `ament_meta_build` by some build system-specific code. In this case, `python-dev` is a package that abstracts away a dependency on CPython development libraries and that will likely be replaced with build system code by each `ament_meta_build` plugin. 

It's worth mentioning that this is an exceptional case. Having each `ament_meta_build` plugin handle a "package" in a custom way is expensive. I think this makes sense for CPython development libraries because there's a large variability across build systems when using them and CPython extensions are ubiquitous. For most cases, it'll be up to the user to make sure a package requirement makes sense for the target build system (e.g. a requirement on `fastrtps` in a `.cbs` file will translate to `find_package(fastrtps REQUIRED)` in CMake -- it is up to the user to make sure that such macro call succeeds and provides the necessary information).